visualizations/frontend/filters/v2/TTagsInput.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+	<div class="t-tags-input max-w-max min-w-full min-h-[43px] relative">
+		<div class="font-bold tracking-widest pb-2" v-if="label">
+			{{ label.toUpperCase() }}
+		</div>
+		<base-multiple-value-input
+			v-if="isSingleTag"
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+			display-key="value"
+			value-key="value"
+		/>
+		<base-tags-input
+			v-else
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+		/>
+	</div>
+</template>
+
+<script>
+	export default {
+		props: {
+			label: {
+				type: String,
+				default: '',
+			},
+			tKeyColumn: {
+				type: String,
+				default: 'KEY',
+			},
+			tValueColumn: {
+				type: String,
+				default: 'VALUE',
+			},
+			canReloadOnOpen: {
+				type: Boolean,
+				default: false,
+			},
+			isSingleTag: {
+				type: Boolean,
+				default: false,
+			},
+			placeholder: {
+				type: String,
+				default: '',
+			}
+		},
+		data: () => ({
+			is_filter: true,
+			selectedTags: [],
+			initialized: false,
+			tempTags: [], // base-multiple-value-input does not show selected items if `tags` is empty because data is fetched on open.
+		}),
+		computed:{ 
+			tags() {
+				const items = [];
+
+				if (this.rows.length) {
+					for (let row of this.rows) {
+						const key = row[this.tKeyColumn];
+						const value = row[this.tValueColumn];
+						if (key && value) {
+							items.push({
+								key: key.value,
+								value: value.value,
+							})
+						}
+					}
+				}
+				return items;
+			},
+			firstKey() {
+				// Used for single tag
+				if (this.tags.length) {
+					return this.tags[0].key;
+				} else if (this.tempTags) {
+					return this.tempTags[0].key;
+				}
+				return null;
+			}
+		},
+		keyColumn() {
+			return this.tKeyColumn ? this.tKeyColumn : this.findColumnByTag('keys');
+		},
+		valueColumn() {
+			return this.tValueColumn ? this.tValueColumn : this.findColumnByTag('values');
+		},
+		methods: {
+			onVisualizationInit() {
+				const initial_value = this.getFilterValue("selected_items");
+				const selectedTags = [];
+				const tempTags = [];
+
+				if (initial_value) {
+					const urlTags = initial_value.split('|');
+					for (let item of urlTags) {
+						const splitPair = item.split('_');
+						const key = splitPair.length > 1 ? splitPair[1] : splitPair[0];
+						const value = splitPair[0];
+
+						if (key && value) {
+							if (this.isSingleTag) {
+								selectedTags.push(value);
+								tempTags.push({ key, value })
+							} else {
+								selectedTags.push({ key, value });
+							}
```

Comment:
Shouldn't we be assigning `tempTags` in both cases? In both cases we are binding data like this:
```js
    :data="tags.length ? tags : tempTags"
```
I guess, but I may be missing something, that `tempTags` is used when there is not `tags`, and `tags` are missing when there is no `rows` (fetched data). So I guess that `tempTags` should be assigned in both cases.

Reply:
Temp tags takes values from url as `tags` on initialization, this hack was only required because the tags component by product cl does not show selected items until it has a array of objects available as `data`, so until we actually click on filter and load, nothing appears as selected.

Reply:
Yes but we are only applying this hack to `singleTag` components, is that right?
I'm asking because we are doing the same for `data` in both components:
```vue
		<base-multiple-value-input
			v-if="isSingleTag"
			v-model="selectedTags"
			selection-only
			:data="tags.length ? tags : tempTags"
			:loading="loading"
			:placeholder="placeholder"
			@opened="handleOpen"
			@update="handleUpdate"
			display-key="value"
			value-key="value"
		/>
		<base-tags-input
			v-else
			v-model="selectedTags"
			selection-only
			:data="tags.length ? tags : tempTags"
			:loading="loading"
			:placeholder="placeholder"
			@opened="handleOpen"
			@update="handleUpdate"
		/>
```

Reply:
Ahh I see, my bad, checking again.

Reply:
@egcastro only `base-multiple-value-input` needs `tempTags`, because it cannot show selected properties until it has array of objects as data but `base-tags-input` does not need it, this is a weird limitation in base component, also temp tags should not be assigned as main variable for either of these, it does not get updated after initial call, as soon as `tags` gets data from backend, that should be used instead.